### ISSUE TITLE
Enhancement: new floating window should not dismiss the preselection

### ISSUE THREAD

**Issue by fxdave** [baskerville/bspwm#1103]
Sometimes when I want to open a new app, I use some intermediate window. This window is floating by default. Preselection, however, should only consider tiling windows.

Current behavior:
Floating window dismisses the preselection:

Excepted behavior:
Open a floating window. The preselection remains untouched.

**Comment by ortango** [comment]
>Preselection, however, should only consider tiling windows.

why? the window needs to be placed in the tree somewhere.

**Comment by fxdave** [comment]
> why? the window needs to be placed in the tree somewhere.

Yes, place somewhere, but it doesn't matter for floating windows.
I would just close them anyway.

Use case:
 - I preselect where should the terminal be.
 - I press `super + p ; o` # for project open
   - It opens a Zenity floating window to ask which project should be opened.
   - I select the project
   - The prompt closes
   - The script opens the terminal in the project root.

**Comment by ortango** [comment]
that seems frustrating.

you could almost solve the issue easily in an external rule script

    eval $4
    [ "$state" == "floating" ] && echo node=@/

but that will not work if there is only one node on the desktop.

another option might be to use a receptacle in this situation instead of preselection.

as far as bspwm goes - it looks like insert_node easily enough to avoid your issue - but i still wouldn't think it would make sense for a blanket rule for floating windows to do this - maybe for windows with WM_TRANSIENT_FOR / dialog windows?

**Comment by fxdave** [comment]
> maybe for windows with WM_TRANSIENT_FOR / dialog windows?
Yeah, that would be a wise choice.
Thanks
